scripts/analysis.py

```diff
@@ -250,229 +116,198 @@ def get_sim_time_duration(cursor):
-    else:
-        multi_line_plot(waste_dict, timestep,
-                        'Years', 'Mass [kg]',
-                        title, outputname, init_year)
+"""GETTERS"""
 
 
-def commodity_from_facility(cursor, facility, commodity):
-    """ Returns timeseries of commodity outflux from facility
+def snf(cursor):
+    """returns a dictionary of isotopics in sink at the end of simulation
 
     Parameters
     ----------
     cursor: sqlite cursor
         sqlite cursor
-    facility: str
-        name of facility type
-    commodity: list
-        list of commodities
 
     Returns
     -------
-    dictionary of timeseries of mass outflux of commodity from facility
+    snf_dict: dictionary
+        dirctionary with key: isotope, and value: mass
     """
+    sink_id = get_agent_ids(cur, 'sink')
 
-    cur = cursor
-    init_year, init_month, duration, timestep = get_sim_time_duration(cur)
-    commodity_dict = collections.OrderedDict()
-    for x in commodity:
-        y = ['"' + x + '"']
-        resources = cur.execute(exec_string(y, 'commodity',
-                                            'quantity, senderid, time')).fetchall()
-
-        timeseries = get_from_facility(cur, facility, duration, resources)
-        commodity_dict[x] = timeseries
+    # get list of sum(quantity) and qualid for snf
+    snf_inventory = cursor.execute(exec_string(sink_id,
+                                               'transactions.receiverId',
+                                               'sum(quantity), qualid') +
+                                   ' GROUP BY qualid').fetchall()
+    compositions = cursor.execute('SELECT qualid, nucid, massfrac '
+                                  'FROM compositions').fetchall()
+    snf_dict = collections.defaultdict(float)
+    for comp in compositions:
+        for num in snf_inventory:
+            if num[0] == comp[0]:
+                snf_dict[comp[1]] += num[1] * comp[2]
 
-    return commodity_dict
+    return snf_dict
 
 
-def get_from_facility(cursor, facility, duration, resources):
-    """ Returns timeseries list of quantity out of facility type
+def get_isotope_transactions(resources, compositions):
+    """Creates a dictionary with isotope name, mass, and time
 
     Parameters
     ----------
-    cursor: sqlite cursor
-        sqlite cursor
-    facility: str
-        name of facility type
-    duration: int
-        duration of simulation
-    resrources: list
-        list of lists, of quantity, senderid and time
+    resources: list
+        resource data from the resources table
+        list[0]: time
+        list[1]: sum(quantity)
+        list[2]: qualid
+    compositions: list
+        composition data from the compositions table
+        list[0]: qualid
+        list[1]: nucid
+        list[2]: massfrac
 
     Returns
     -------
-    list
-    timeseries list of mass outflux from facility type
+    transactions: dictionary
+        dictionary with keys as isotope and value as
```

Comment:
I changed them as you've suggested in the previous comment by using "key=isotope, and value= lists of tuples..."